src/main/java/teammates/common/util/AdminLogQuery.java

```diff
@@ -0,0 +1,175 @@
+     */
+    private void setEndTime(Long endTimeParam) {
+        endTime = endTimeParam;
+    }
+    
+    /**
+     * Gets end time of the query.
+     */
+    public Long getEndTime() {
+        return endTime;
+    }
+    
+    /**
+     * Gets versions used in query.
+     */
+    public List<String> getVersionsToQuery() {
+        return versionList;
+    }
+    
+    /**
+     * Selects versions for query. If versions are not specified, it will return 
+     * default versions used for query.
+     */
+    private List<String> getVersionIdsForQuery(List<String> versions) {
+        boolean isVersionSpecifiedInRequest = (versions != null && !versions.isEmpty());
+        if (isVersionSpecifiedInRequest) {
+            return versions;
+        }
+        return getDefaultVersionIdsForQuery();
+    }
+    
+    /**
+     * Gets a list of versions, including the current version and 5 preceding versions (if available).
+     * @return a list of default versions for query.
+     */
+    private List<String> getDefaultVersionIdsForQuery() {
+        GaeVersionApi adminApi = new GaeVersionApi();
+        List<Version> versionList = adminApi.getAvailableVersions();
+        Version currentVersion = adminApi.getCurrentVersion();
+        
+        List<String> defaultVersions = new ArrayList<String>();
+        try {
+            int currentVersionIndex = getCurrentVersionIndex(versionList, currentVersion);
+            defaultVersions = getNextFewVersions(versionList, currentVersionIndex);
+        } catch (InvalidParametersException e) {
+            defaultVersions.add(currentVersion.toStringForQuery());
+            Utils.getLogger().severe(e.getMessage());
+        }
+        return defaultVersions;
+    }
+
+    /**
+     * Finds at most MAX_VERSIONS_TO_QUERY nearest versions.
+     * @param currentVersionIndex starting position to get versions to query
+     */
+    private List<String> getNextFewVersions(List<Version> versionList, int currentVersionIndex) {
+        int endIndex = Math.min(currentVersionIndex + MAX_VERSIONS_TO_QUERY, versionList.size());
+        List<Version> versionSubList = versionList.subList(currentVersionIndex, endIndex);
+        List<String> versionListInString = new ArrayList<String>();
+        for(Version version : versionSubList) {
+            versionListInString.add(version.toStringForQuery());
+        }
+        return versionListInString;
+    }
+
+    /**
+     * Finds the index of the current version in the given list.
+     * @throws InvalidParametersException when the current version is not found
+     */
+    private int getCurrentVersionIndex(List<Version> versionList, Version currentVersion) 
+                    throws InvalidParametersException {
+        int versionIndex = versionList.indexOf(currentVersion);
+        if (versionIndex != -1) {
+            return versionIndex;
+        }
+        throw new InvalidParametersException("The current version is not found!");
+    }
```

Comment:
This too can go to `Version` class?. It's dealing with versions and not accessing internal data of this class.